Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3800 @@
+import demistomock as demisto
+from CommonServerPython import *  # noqa #! pylint: disable=unused-wildcard-import
+from typing import Dict, Any
+import urllib3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+""" CONSTANTS """
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+PORT_CREDENTIALS = "9182"
+PORT_AUTH = "9183"
+DEFAULT_TIMEOUT = 600
+DEFAULT_INTERVAL = 60
+CLIENT_ID = ""
+CLIENT_SECRET = ""
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):
+    def __init__(
+        self,
+        server_url,
+        verify,
+        proxy,
+        headers,
+        auth,
+        organization_params,
+        client_id_param,
+        client_secret_param,
+    ):
+        super().__init__(base_url=server_url, verify=verify, proxy=proxy, headers=headers, auth=auth)
+        self.organization_params = organization_params
+        self.client_id_param = client_id_param
+        self.client_secret_param = client_secret_param
+
+    # HEADING: """ REQUEST FUNCTIONS """
+
+    def access_token_request(self, username: str, password, client_id: str, client_secret: str):
+        request_body = {
+            "client_id": f"{client_id}",
+            "client_secret": f"{client_secret}",
+            "username": f"{username}",
+            "password": f"{password}",
+            "grant_type": "password",
+        }
+
+        response = self._http_request(
+            "POST",
+            url_suffix="auth/token",
+            headers=self._headers,
+            json_data=request_body,
+            ok_codes=(200, 201),
+        )
+        return response
+
+    def refresh_token_request(self, client_id: str, client_secret: str, refresh_token: str):
+        request_body = {
+            "client_id": f"{client_id}",
+            "client_secret": f"{client_secret}",
+            "grant_type": "refresh_token",
+            "refresh_token": f"{refresh_token}",
+        }
+
+        response = self._http_request(
+            "POST",
+            url_suffix="auth/refresh",
+            headers=self._headers,
+            json_data=request_body,
+        )
+
+        return response
+
+    def auth_credentials_request(self, access_token: str, token_name: str, token_description: str):
+        request_body = {
+            "name": f"{token_name}",
+            "description": f"{token_description}",
+            "expires_at": "",
+            "client_secret_expires_at": "",
+            "max_access_tokens": 10,
+            "max_access_tokens_per_user": 99,
+            "access_token_validity": 900,
+            "refresh_token_validity": 86400,
+            "allowed_grant_types": ["password", "refresh_token", "client_credentials"],
+            "allowed_source_client_address": {
+                "source_type": "ANYWHERE",
```

Comment:
Lets add a comment here/ document in confluence the grants and source type available.